docs/application/native/guides/text-input/capi-ui-gesture.md

```diff
@@ -0,0 +1,248 @@
+
+   > **Note**
+   >
+   > Please set the error callback function before calling `hand_gesture_start_recognition()`.
+
+    ```
+    /* Callback */
+    void
+    error_cb(hand_gesture_h gesture_h, hand_gesture_error_e reason, const char* msg, void* user_data)
+    {
+        /* Your code */
+    }
+
+    /* Set */
+    void
+    set_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_set_error_cb(gesture_h, error_cb, NULL);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+    /* Unset */
+    void
+    unset_error_cb(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_unset_error_cb(gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. Recognition result
+
+   When the hand gesture is recognized, the recognized result is delivered through the `hand_gesture_recognition_cb()`.
+
+   You can [set the recognition result callback function using `hand_gesture_start_recognition()`](#start_stop).
+
+
+<a name="start_stop"></a>
+## Start and Stop recognizing the gesture
+
+1. To start recognizing the gesture, use the `hand_gesture_start_recognition()` function.
+
+   When you start the gesture recognition, you should input the gesture type targeted to be recognized. Before calling `hand_gesture_start_recognition()`, please check whether the target gesture type is supported or not with `hand_gesture_is_supported_type()`.
+
+   After calling `hand_gesture_start_recognition()`, the recognition results will be coming through the callback function `hand_gesture_recognition_cb()`.
+   When the callback function is invoked, you can handle the recognition results by corresponding to some operations.
+   For example, when `HAND_GESTURE_WRIST_UP` is occurred, you can turn on the screen of the wearable device.
+
+    ```
+    void
+    recognition_cb(hand_gesture_h gesture_h, hand_gesture_type_e gesture, double timestamp, hand_gesture_error_e error, void* user_data)
+    {
+        /* Your code */
+    }
+
+    void
+    start_recognition(hand_gesture_h gesture_h)
+    {
+        int ret;
+        bool is_supported = false;
+        ret = hand_gesture_is_supported_type(gesture_h, HAND_GESTURE_WRIST_UP, &is_supported);
+        if (HAND_GESTURE_ERROR_NONE == ret && true == is_supported)
+        {
+            ret = hand_gesture_start_recognition(gesture_h, HAND_GESTURE_WRIST_UP, recognition_cb, NULL);
+            if (HAND_GESTURE_ERROR_NONE != ret)
+                /* Error handling */
+        }
+    }
+    ```
+
+2. To stop recognizing the gesture, use the `hand_gesture_stop_recognition()` function.
+
+    ```
+    void
+    stop_recognition(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_stop_recognition(gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+
+
+## Related Information
```

Comment:
Please change to:

## Related information

Reply:
Done.